app/controllers/storage_locations_controller.rb

```diff
@@ -217,7 +217,7 @@ def load_storage_location
   end
 
   def check_read_permissions
-    render_403 unless can_read_storage_location?(@storage_location)
+    render_403 unless can_read_storage_location?(@storage_location) && @storage_location.container
```

Comment:
why is it here?

Reply:
I think instead, if it's not a container, we should redirect to the index with parent_id set with its id, probably just with a guard clause on top of the show action.